source/docs/advanced/graphql/slim-down-resolvers-with-loaders.md

```diff
@@ -0,0 +1,351 @@
+  then be the only part of your code to adapt to this complexity.
+</blockquote>
+
+## Refactoring to loaders
+
+We will go through the process of extracting business logic from resolvers to a
+loader and discover the different tools provided by Front-Commerce to achieve
+this.
+
+### Initial resolvers
+
+For this example, we will reuse loaders from the example used in the
+[Extend the GraphQL schema](/docs/essentials/extend-the-graphql-schema.html#Implement-resolvers)
+guide.
+
+Here is the initial code we will refactor:
+
+```js
+// my-module/server/modules/clicks-counters/resolvers.js
+const counters = new Map();
+
+const currentValueOf = sku => {
+  return new Promise(resolve => {
+    setTimeout(() => {
+      const currentValue = counters.get(sku) || 0;
+      resolve(currentValue);
+    }, Math.random() * 3000);
+  });
+};
+
+module.exports = {
+  Product: {
+    clicksCounter: ({ sku }) => currentValueOf(sku)
+  },
+
+  Mutation: {
+    incrementProductCounter(_, { sku, incrementValue = 1 }) {
+      return currentValueOf(sku)
+        .then(currentValue => counters.set(sku, currentValue + incrementValue))
+        .then(() => ({
+          success: true
+        }));
+    }
+  }
+};
+```
+
+### Extract code in a loader
+
+Let’s start by creating a new `loader.js` file with our loader module.
+
+```js
+// my-module/server/modules/clicks-counters/loader.js
+const counters = new Map();
+
+const currentValueOf = sku => {
+  return new Promise(resolve => {
+    setTimeout(() => {
+      const currentValue = counters.get(sku) || 0;
+      resolve(currentValue);
+    }, Math.random() * 3000);
+  });
+};
+
+const incrementValueOf = (sku, increment) => {
+  return currentValueOf(sku).then(currentValue =>
+    counters.set(sku, currentValue + increment)
+  );
+};
+
+export default {
+  loadBySku: sku => currentValueOf(sku),
+  incrementBySku: (sku, increment) => incrementValueOf(sku, increment)
+};
+```
+
+You can then update the resolvers to use this loader:
+
+```diff
+// my-module/server/modules/clicks-counters/resolvers.js
+-const counters = new Map();
++import CounterLoader from "./loader";
+
+-const currentValueOf = sku => {
+-  return new Promise(resolve => {
+-    setTimeout(() => {
+-      const currentValue = counters.get(sku) || 0;
+-      resolve(currentValue);
+-    }, Math.random() * 3000);
+-  });
+-};
+-
+module.exports = {
+  Product: {
+-    clicksCounter: ({ sku }) => currentValueOf(sku)
++    clicksCounter: ({ sku }) => CounterLoader.loadBySku(sku)
+  },
+
+  Mutation: {
+    incrementProductCounter(_, { sku, incrementValue = 1 }) {
+-      return currentValueOf(sku)
+-        .then(currentValue => counters.set(sku, currentValue + incrementValue))
++      return CounterLoader.incrementBySku(sku, incrementValue)
+        .then(() => ({
+          success: true
+        }));
+    }
+  }
+};
+```
+
+### Using GraphQL context for Dependency Injection
+
+In its current state, resolvers are tightly coupled to the current
```

Comment:
It's not really clear what current state you are refering to. I'd prefer something like

```suggestion
However, in the above example, resolvers are tightly coupled to the current
```